### ISSUE TITLE
Tutorial ends at Calling Open Metadata and Governance Services, Jupyter labs ends up with errors

### ISSUE THREAD

**Issue by MarkizaSckuza** [odpi/egeria#5606]
I'm trying to run Egeria locally, using tutorial.
I was able to build (version 3.1-SNAPSHOT), configure and start OMAG server, as described [here](https://egeria.odpi.org/open-metadata-resources/open-metadata-tutorials/omag-server-tutorial/task-starting-omag-server.html).
But the next step is **Calling Open Metadata and Governance Services**, but tutorial for it is [empty](https://egeria.odpi.org/open-metadata-resources/open-metadata-tutorials/omag-server-tutorial/task-calling-omag-apis.html)
So, where can I find next steps?

Also, I successfully installed helm chart with Coco Pharmaceuticals and ran notebooks in the next order:
1. egeria-server-config.ipynb
2. egeria-server-start.ipynb
3. administration-labs/understanding-server-config.ipynb
4. administration-labs/managing-servers.ipynb
5. asset-management-labs/building-a-data-catalog.ipynb - at this notebook, 11th item returned:

> Review of the assets so far:
 Asset 1 GUID is: <unknown>
 Asset 2 GUID is: 1bc534e5-b8ca-4f23-bcf7-4faa0ac08f48
 Asset 3 GUID is: 49e27f02-4f9f-4b14-bc62-81264a67d6a8

and 12th said:

> No comment added
OMAG-REPOSITORY-HANDLER-500-002 The Open Metadata Service Asset Consumer OMAS is not able to set the Anchors classification on a new entity of type Anchors during method addCommentToAsset. The resulting exception was org.odpi.openmetadata.frameworks.connectors.ffdc.PropertyServerException with error message OMAG-GENERIC-HANDLERS-500-009 An anchor GUID of <unknown> has been passed to local method setAnchors by the Asset Consumer OMAS service through method addCommentToAsset
 * The server was attempting to add Anchors classifications to a collection of metadata instances that are logically part of the same object.  This classification is used to optimize the retrieval and maintenance of complex objects.  It is optional function.  The server continues to process the original request which will complete successfully unless something else goes wrong.
 * No specific action is required.  This message is to highlight that the retrieval and management of metadata is not optimalbecause none of the repositories in the cohort support the Anchors classification.  To enable the optimization provided through the Anchors classification, add an Egeria native metadata server to the cohort.  This will provide the support for the Anchors classification.
 
and than 13th:

> No comments returned
OMAG-REPOSITORY-HANDLER-404-007 The Asset entity with unique identifier <unknown> is not found for method getAssetComments of access service Connected Asset Services in open metadata server cocoMDS2, error message was: OMRS-REPOSITORY-404-002 The entity identified with guid <unknown> passed on the getEntityDetail call is not known to the open metadata repository cocoMDSx
 * The system is unable to update information associated with the entity because none of the connected open metadata repositories recognize the entity's unique identifier.
 * The unique identifier of the entity is supplied by the caller.  Verify that the caller's logic is correct, and that there are no errors being reported by the open metadata repository. Once all errors have been resolved, retry the request.

at 14th I've got an error:

> 
> <unknown>
>
> ---------------------------------------------------------------------------
> TypeError                                 Traceback (most recent call last)
> /tmp/ipykernel_127/2114521567.py in <module>
>       4 
>       5 print(asset1guid)
> ----> 6 replyGUID = addReplyToAssetComment(cocoMDS1Name, cocoMDS1PlatformName, cocoMDS1PlatformURL, petersUserId, asset1guid, commentGUID, commentText, commentType, isPublic)
>       7 
>       8 print (" ")
> 
> /tmp/ipykernel_127/3860355654.py in addReplyToAssetComment(serverName, serverPlatformName, serverPlatformURL, userId, requestType, commentGUID, commentText, commentType, isPublic)
>      36 def addReplyToAssetComment(serverName, serverPlatformName, serverPlatformURL, userId, requestType, commentGUID, commentText, commentType, isPublic):
>      37     assetConsumerURL = serverPlatformURL + '/servers/' + serverName + '/open-metadata/access-services/asset-consumer/users/' + userId
> ---> 38     commentReplyURL = assetConsumerURL + '/assets/' + requestType + '/comments/' + commentGUID + '/replies'
>      39     print (" ")
>      40     print ("POST " + commentReplyURL)
> 
> TypeError: can only concatenate str (not "NoneType") to str

Also, I wasn't able to use UI. 
I ran `ui-labs/ui-asset-search.ipynb` Jupyter notebook and received UI url (egeria-nginx:443), port forwarded it from Kubernetes, authenticated into it (by the way, using `admin` `admin` creds, not the ones, written in notebook - cause with them I receive `You have entered incorrect username or password.`), opened `repository-explorer` and received an error:

> Error occurred: The repository could not be reached, please check the server name and URL root and verify that the server is running 

My UI does not look like at pictures in notebook:

![image](https://user-images.githubusercontent.com/8726391/130605872-5dc93867-403b-4cc6-a606-fe49b8c5429a.png)

**Comment by planetf1** [comment]
Thanks for the report.

On the 'building a data catalog' notebook. I was able to run this through from beginning to end using the helm chart.

Can I check - are you running the new instructions at https://odpi.egeria.io/egeria-charts (the docs will be moving again shortly as we're doing some refactoring).

Basically:
```
helm repo add egeria https://odpi.github.io/egeria-charts
helm repo update egeria
helm install egeria/odpi-egeria-lab
```

The latest released chart is setup to use the 3.0 egeria image
```
$ helm search repo egeria                                                                                                                                 [9:18:00]
NAME                    CHART VERSION   APP VERSION     DESCRIPTION                           
egeria/egeria-base      3.0.1                           Egeria simple deployment to Kubernetes
egeria/odpi-egeria-lab  3.0.1                           Egeria lab environment           
```

You can also see any development versions with:
```
$ helm search repo egeria --devel                                                                                                                         [9:19:28]
NAME                    CHART VERSION   APP VERSION     DESCRIPTION                           
egeria/egeria-base      3.0.1                           Egeria simple deployment to Kubernetes
egeria/odpi-egeria-lab  3.0.1                           Egeria lab environment    
```
but there are no later versions yet.

Will check my images and see if I can reproduce

**Comment by planetf1** [comment]
I also repeat the run-through on the notebook running Egeria locally (this time 3.1-SNAPSHOT with master as of this morning).

On the Polymer UI, the passwords did change in the recent release. see https://github.com/odpi/egeria/blob/master/release-notes/release-notes-3.0.md - they are now 'secret'. However the function of that UI is not fully tested in the 'lab' environment. See https://github.com/odpi/egeria-charts/issues/19 . 

Also are you able to use the react UI (you will need to expose the '*presentation' service - defaults to port 8091 & go to https://address:8091/coco ). Login as garygeeke/admin and try dino/tex/rex there. They are running later versions. Or do you get the same error? One difference is the way the servers being targetted are configured/constrained. You select from a drop down list.

That being said I checked the polymer UI, and it is configured to point to 'cocoMDS4' on lab-datalake:9443 . That is fine, except that cocoMDS4 isn't running a repository. If our 'enterprise connector' is used by the egeria-ui I believe it should work. So there is an issue here, but it's tied up in the dev/test/demo process of that ui. In the react UI we don't include cocoMDS4 in the list. I'd have to check a few more things to see if there is anything wrong here, or is it just the UI issue

In summary I think there are a few things to persue

a) Your environment. Why are you seeing different to what I do when running either helm or locally. Let's continue that here. 

b) The integration of the egeria UI. At the level of demo integration that is referenced in https://github.com/odpi/egeria-charts/issues/19 . I think this includes the use of cocoMDS4. Whether this is important to you may depend on your requirements. for dev type interaction I'd recommend the react UI, whilst for business access the egeria ui. Both are still very much in development.

**Comment by planetf1** [comment]
I suggest that since you are already deploying to k8s, ensuring you can run the latest released version clean is a good starting point (local can be checked later).

* check images are current

Your environment is probably going to cache docker images. You can try deploying the chart by creating a file '~/lab.yaml' (for example) with the contents
```
imageDefaults:
    pullPolicy: Always
```
We'll use this file to override settings in the chart. In this case we change the policy to always pull a fresh container image when starting the pod. Inefficient, but will make sure nothing is cached.

Then deploy the chart with `helm install egeria/odpi-egeria-lab -f ~/etc/lab.yaml`

Any difference here?

**Comment by planetf1** [comment]
For your helm chart deployed to k8s, can you also check the logs reported by your pods for any odd errors? (ie kubectl logs <pod>)

**Comment by planetf1** [comment]
Hi @MarkizaSckuza 

Glad to see the chart is now working for you (also, we just released the final 3.1 -- but there were no code changes) - though I don't know why it failed before.

The helm chart is a quick way to get the lab running, but it's really just a starting point. We haven't yet added any examples around Lineage to the coco environment, or the 'egeria-base' deployment. We hope to in future. For experimenting you could write your own notebook(s) [save them externally] if that's useful, or you could save some postman scripts to send in to egeria to do additional configuration. For a 'real' deployment you could use the charts as an example to begin with.

There are a few different areas of activity around lineage. The egeria open lineage service I think you are referring to is at https://github.com/odpi/egeria/tree/master/open-metadata-implementation/governance-servers/open-lineage-services - but you'll need some guidance to get started (I haven't used it myself). 

I would suggest either
 * joining our community call or developers call (see https://odpi.github.io/egeria-docs/guides/community/)
 * open a specific issue on help to get started with Open Lineage Services (may be clearer now that we have your initial issue resolved) & we can assign to the right experts in that area.
 
 In addition if you still have any issues running egeria locally let me know here.

**Comment by MarkizaSckuza** [comment]
> I would suggest either
> 
> * joining our community call or developers call (see https://odpi.github.io/egeria-docs/guides/community/)
> * open a specific issue on help to get started with Open Lineage Services (may be clearer now that we have your initial issue resolved) & we can assign to the right experts in that area.

Thank you, @planetf1

Closing this issue, will open the new one for Open Lineage Services
